src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.ObsoletedInOSPlatformTests.cs

```diff
@@ -0,0 +1,457 @@
+    Microsoft.NetCore.Analyzers.InteropServices.PlatformCompatibilityAnalyzer,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.NetCore.Analyzers.InteropServices.UnitTests
+{
+    public partial class PlatformCompatabilityAnalyzerTests
+    {
+        [Fact]
+        public async Task ObsoletedMethodsCalledWarns()
+        {
+            var csSource = @"
+using System;
+using System.Runtime.Versioning;
+using Mock;
+
+public class Test
+{
+    [SupportedOSPlatform(""Linux"")]
+    public void M1()
+    {
+        ObsoletedOnWindows10(); // Shold not warn as only accessible on Linux
+        {|#0:ObsoletedOnLinux4()|}; // This call site is reachable on: 'Linux'. 'Test.ObsoletedOnLinux()' is obsoleted on: 'Linux' 4.1 and later.
+        {|CA1422:ObsoletedOnLinux4AndWindows10()|}; // This call site is reachable on: 'Linux'. 'Test.ObsoletedOnLinux4AndWindows10()' is obsoleted on: 'Linux' 4.1 and later.
+    }
+    
+    [ObsoletedInOSPlatform(""Linux4.1"")]
+    public void ObsoletedOnLinux4() { }
+
+    [ObsoletedInOSPlatform(""Windows10.1.1.1"")]
+    public void ObsoletedOnWindows10() { }
+
+    [ObsoletedInOSPlatform(""Linux4.1"", ""Use Linux4Supported"")]
+    [ObsoletedInOSPlatform(""Windows10.1.1.1"",""Use Windows10Supported"")]
+    public void ObsoletedOnLinux4AndWindows10() { }
+}" + MockObsoletedAttributeCS;
+            await VerifyAnalyzerCSAsync(csSource, VerifyCS.Diagnostic(PlatformCompatibilityAnalyzer.ObsoletedCsReachable).WithLocation(0)
+                .WithArguments("Test.ObsoletedOnLinux4()", "'Linux' 4.1 and later", "'Linux'"));
+
+
+            var vbSource = @"
+Imports System
+Imports System.Runtime.Versioning
+Imports Mock
+Public Class Test
+    <SupportedOSPlatform(""Linux"")>
+    Public Sub M1()
+        ObsoletedOnWindows10()
+        {|#0:ObsoletedOnLinux()|} ' This call site is reachable on: 'Linux'. 'Public Sub ObsoletedOnLinux()' is obsoleted on: 'Linux' 4.1 and later.
+    End Sub
+
+    <ObsoletedInOSPlatform(""Windows10.1.1.1"")>
+    Public Sub ObsoletedOnWindows10()
+    End Sub
+    
+    <ObsoletedInOSPlatform(""Linux4.1"")>
+    Public Sub ObsoletedOnLinux()
+    End Sub
+End Class
+" + MockObsoletedAttributeVB;
+            await VerifyAnalyzerVBAsync(vbSource, VerifyVB.Diagnostic(PlatformCompatibilityAnalyzer.ObsoletedCsReachable).WithLocation(0)
+                .WithArguments("Public Sub ObsoletedOnLinux()", "'Linux' 4.1 and later", "'Linux'"));
+        }
+
+        [Fact]
+        public async Task ObsoletedWithMessageUrlCalledWarns()
+        {
+            var csSource = @"
+using System;
+using System.Runtime.Versioning;
+using Mock;
+
+public class Test
+{
+    public void CrossPlatform()
+    {
+        {|#0:ObsoletedWithMessageAndUrl()|}; // This call site is reachable on all platforms. 'Test.ObsoletedWithMessageAndUrl()' is obsoleted on: 'Windows' 10.1.1.1 and later, Use other method instead, http://www/look.for.more.info.
```

Comment:
The problem is we are separating the platforms with comma, when the API annotated with multiple platforms, for example: `This call site is reachable on all platforms. 'Test.ObsoletedOnWindowsAndLinux()' is obsoleted on: 'Windows', 'Linux' .`

In this case when there is custom message and or Url for a specific platform, those need to be appended to the corresponding platform and having a period will be not appropriate, it would look like this:
`This call site is reachable on all platforms. 'Test.ObsoletedOnWindowsAndLinux()' is obsoleted on: 'Windows', Windows message  http://www.windows.url.com, 'Linux', Linux message http://www.linux.url.com.`

But the custom message also could have a period and could break the structure, I have no good solution here

Reply:
Ah, I didn't think about each platform having its own message/url. This is a bit trickier to pull off, but how about this:

`This call site is reachable on all platforms. 'Test.ObsoletedOnWindowsAndLinux()' is obsoleted on: 'Windows' (Windows message http://www.windows.url.com), 'Linux' (Linux message http://www.linux.url.com).`

You'd need to create resource strings that represent the parenthesis and the custom message/url):
1. `({message} {url})`
2. ({messageOrUrl})`

@gewarren might have other suggestions here too.

Reply:
That sounds good to me!

Reply:
> You'd need to create resource strings that represent the parenthesis and the custom message/url):

Is the parenthesis could be different in other languages? Else string concatenation `$" ({message} {url})"` might be simpler

Reply:
Ended up using resource string 2. `({messageOrUrl})`